### DISCUSSION TITLE
How do I compile a library to run on the android platform?

### DISCUSSION THREAD

**Discussion by zh302674199** [COVESA/dlt-daemon#625] [Q&A] [answered]
1. How to compile a dynamic library that can run on the android system?
2. How to port the project to the android system and run it?

**Comment by minminlittleshrimp** [comment, marked as answer]
Hello @zh302674199 
For question 1, it is out of the scope of DLT, sorry but I would not answer.
For question 2, the project contains Android.bp, this supports Android version > 12 (Android S), since there some big changes in system source dir. DLT is a third paerty vendor module, and it uses Android Soong build system.

To build DLT on Android, please setup the environment using Android build script and jump to root dir of the project, run: mm (make module), further information see:
https://android.googlesource.com/platform/build/+/main/Usage.txt
https://android.googlesource.com/platform/build/soong/+/master

Hope this would help, and, I move this to discussion and close.
